src/fides/api/service/connectors/sql_connector.py

```diff
@@ -199,11 +199,30 @@ def retrieve_data(
         stmt: Optional[TextClause] = query_config.generate_query(input_data, policy)
         if stmt is None:
             return []
+
         logger.info("Starting data retrieval for {}", node.address)
         with client.connect() as connection:
             self.set_schema(connection)
-            results = connection.execute(stmt)
-            return self.cursor_result_to_rows(results)
+            rows = []
+            if query_config.partitioning:
+                partition_clauses = query_config.get_partition_clauses()
+                logger.info(
+                    f"Executing {len(partition_clauses)} partition queries for node '{node.address}' in DSR execution"
+                )
+                for partition_clause in partition_clauses:
+                    logger.debug(
+                        f"Executing partition query with partition clause '{partition_clause}'"
+                    )
+                    existing_bind_params = stmt.compile().params
+                    partitioned_stmt = text(
+                        f"{stmt} AND ({text(partition_clause)})"
+                    ).params(existing_bind_params)
+                    results = connection.execute(partitioned_stmt)
+                    rows.extend(self.cursor_result_to_rows(results))
```

Comment:
This feels like it shouldn't be here yet while it's only supported for bigquery -

Reply:
that's fair, but i'd generally prefer not to create parallel implementations for `retrieve_data` (for maintenance purposes) and i'm not seeing a simple way to refactor this otherwise - but maybe i'm not being creative enough!